pkg/clusterconditions/promql/promql.go

```diff
@@ -98,6 +101,31 @@ func (p *PromQL) Valid(ctx context.Context, condition *configv1.ClusterCondition
 	return nil
 }
 
+// Type CustomClient encapsulates an API client interface.
```

Comment:
Let's call this something more descriptive (`PostProhibitingClient`? `ClientWithoutPost`?), and include the reason why we need this in the client GoDoc (not in the `Do` method's one. Let's mention that downstream client tries POST then fallbacks to GET on not implemented, and that target server maps the initial `POST` to a `create` verb with a standard kube-rbac-proxy logic, and our RBAC should not need `create`.

I think we also want to do this only conditionally, when we know we are in Hypershift. There's no reason to mess with POSTs on standalone. So I'd put just the capability in this PR, and don't use it. Then once the conditionals in the Hypershift PR are in, let's add a "if we are in hypershift, use this client, otherwise use the dfault one" logic.